File: 2026-08/src/day_03/prms_84512/Solution.java
package day_03.prms_84512;

import java.util.*;

class Solution {

    static List<String> list = new ArrayList<>();
    static char[] arr = {'A', 'E', 'I', 'O', 'U'};
    static char[] sel = new char[5];

    public int solution(String word) {
        for (int i = 1; i <= 5; i++) {
            dfs(0, i);
        }
        list.sort(Comparator.naturalOrder());

        for (int i = 0; i < list.size(); i++) {
            if (list.get(i).equals(word)) {
                return i + 1;
            }
        }

        return 0;
    }

    static void dfs(int idx, int len) {
        if (idx == len) {
            StringBuilder sb = new StringBuilder();
            for (int i = 0; i < len; i++) {
                sb.append(sel[i]);
            }
            list.add(sb.toString());
            return;
        }

        for (char c : arr) {
            sel[idx] = c;
            dfs(idx + 1, len);
        }
    }
}

File: 2026-08/src/day_03/prms_84512/Solution.cpp
#include <algorithm>
#include <string>
#include <vector>

using namespace std;

vector<string> v;
string arr = "AEIOU";
string sel(5, ' ');

void dfs(int idx, int len) {
    if (idx == len) {
        v.push_back(sel.substr(0, len));
        return;
    }

    for (char c : arr) {
        sel[idx] = c;
        dfs(idx + 1, len);
    }
}

int solution(string word) {
    for (int i = 1; i <= 5; i++) {
        dfs(0, i);
    }
    sort(v.begin(), v.end());

    for (int i = 0; i < v.size(); i++) {
        if (v[i] == word) {
            return i + 1;
        }
    }

    return 0;
}
